src/map/status.cpp

```diff
@@ -14923,6 +14975,24 @@ TIMER_FUNC(status_change_timer){
 		}
 		sc_timer_next(500 + tick);
 		return 0;
+	case SC_TALISMAN_OF_PROTECTION:
+		if( sce->val4 >= 0 ){
+			// Get the original caster
+			map_session_data* ssd = map_id2sd( sce->val2 );
+
+			// If the caster is offline, dead, on another map or
+			// if the target is not a player or is in another party
+			if( ssd == nullptr || status_isdead( &ssd->bl ) || ssd->bl.m != bl->m || sd == nullptr || ssd->status.party_id != sd->status.party_id ){
+				// End the status change
+				sce->val4 = 0;
+				break;
+			}
+
+			int hp = skill_calc_heal( &ssd->bl, bl, SOA_TALISMAN_OF_PROTECTION, sce->val1, true );
+
+			status_heal( bl, hp, 0, 0, 2 );
```

Comment:
Heal effect (not 0 HP) is also displayed on official server when player hp is hpmax.